artio-core/src/main/java/uk/co/real_logic/artio/engine/logger/FixArchivePrinter.java

```diff
@@ -387,8 +405,8 @@ private static void print(
     {
         final MessageStatus status = message.status();
         final long timestamp = message.timestamp();
-        final String body = message.body();
-        System.out.printf("%1$20s: %2$s (%3$s)%n", timestamp, body, status);
+        final String body = pipeDelimiter ? message.body().replace('\u0001', '|') : message.body();
```

Comment:
I don't know how far you would like to take this, but `SOH` could also appear in `data` fields, so you might end up with pipe symbols in fields that should not be replaced.
But I think this approach is good enough for the majority of cases.

Reply:
Good point, thanks. I think this is good enough for now, `fix.core.debug.separator` doesn't handle data fields either.